include/itkCartesianToPolarTransform.h

```diff
@@ -1,184 +1,184 @@
-    }
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkCartesianToPolarTransform_h
+#define itkCartesianToPolarTransform_h
+
+#include <iostream>
+#include "itkTransform.h"
+#include "itkExceptionObject.h"
+#include "itkMatrix.h"
+
+namespace itk
+{
+
+/** \class CartesianToPolarTransform
+ *
+ * \brief Polar transformation of a vector space (e.g. space coordinates).
+ *
+ * Transforms first two coordinates form cartesian coordinates  to polar
+ * coordinates <alpha,radius>. Other dimensions are left unchanges. In fact
+ * this is generalized cylindric transform:
+ * \f[          r = \sqrt{ x_0^2 + x_1^2 } \f]
+ * \f[          \alpha = \left\{ \begin{array}{ll}
+ * arccos( \frac{x_0}{r} ) & \mbox{$x_1 >= 0$} \\
+ * \mbox{2 \pi} - arccos( \frac{x_0}{r} ) & \mbox{$x_1 < 0$}
+ * \end{array}\right. \f]
+ * \f[          x_n = x_n, \mbox{n >= 2} \f]
+ *
+ *
+ * \par
+ * Center of the polar transform is can be specified with SetCenter().
+ * The default is center of coordinate system < 0, 0 >.
+ *
+ * Dimension must be at least 2.
+ *
+ * \author Jakub Bican, Department of Image Processing, Institute of Information Theory and Automation, Academy of Sciences of the Czech Republic.
+ *
+ * \ingroup Transforms
+ * \ingroup PolarTransform
+ */
+template <
+    typename TParametersValueType=double,          // Data type for scalars (float or double)
+    unsigned int NDimensions=3>        // Number of dimensions
+class ITK_TEMPLATE_EXPORT CartesianToPolarTransform :
+          public Transform< TParametersValueType, NDimensions, NDimensions >
+{
+public:
+  /** Standard class typedefs. */
+  typedef CartesianToPolarTransform                                   Self;
+  typedef Transform< TParametersValueType, NDimensions, NDimensions > Superclass;
+  typedef SmartPointer< Self >                                        Pointer;
+  typedef SmartPointer< const Self >                                  ConstPointer;
+
+  /** New macro for creation of through the object factory.*/
+  itkNewMacro( Self );
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro( CartesianToPolarTransform, Transform );
+
+  /** Dimension of the domain space. */
+  static constexpr unsigned int SpaceDimension = NDimensions;
```

Comment:
Do we want `static` here?

Reply:
That what was there before.  Perhaps not, but I'm not trying to maintain this code at the same level as ITK proper.   Feel free to make changes as you see fit.

PS:  Thanks for you reviews.